ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Praos/Translate.hs

```diff
@@ -18,21 +18,35 @@ import qualified Cardano.Protocol.TPraos.Rules.Prtcl as SL
 import qualified Cardano.Protocol.TPraos.Rules.Tickn as SL
 import           Data.Coerce (coerce)
 import qualified Data.Map.Strict as Map
+import           Ouroboros.Consensus.Block
 import           Ouroboros.Consensus.Protocol.Praos (ConsensusConfig (..),
                      Praos, PraosParams (..), PraosState (..),
-                     Ticked (TickedPraosLedgerView))
+                     Ticked (TickedPraosLedgerView),
+                     tickedPraosStateChainDepState)
 import           Ouroboros.Consensus.Protocol.Praos.Views
                      (LedgerView (lvMaxBodySize, lvMaxHeaderSize, lvProtocolVersion))
 import qualified Ouroboros.Consensus.Protocol.Praos.Views as Views
-import           Ouroboros.Consensus.Protocol.TPraos (TPraos, TPraosParams (..),
-                     TPraosState (tpraosStateChainDepState, tpraosStateLastSlot))
+import           Ouroboros.Consensus.Protocol.TPraos (TPraos, TPraosParams (..))
 import qualified Ouroboros.Consensus.Protocol.TPraos as TPraos
 import           Ouroboros.Consensus.Protocol.Translate (TranslateProto (..))
 
 {-------------------------------------------------------------------------------
   Translation from transitional Praos
 -------------------------------------------------------------------------------}
 
+instance (c1 ~ c2) => TranslateProto (TPraos c1) (TPraos c2) where
```

Comment:
I wonder if the error message would actually be easier to understand if the head was `TranslateProto (TPraos c) (TPraos c)`?

And same for `Praos` below.

Reply:
I would have to play around to find this out, I *think* the way how it is written here can sometimes be beneficial because it causes instances selection to pick this instance even if it does not (yet) know `c1 ~ c2` (a similar example is discussed [here](https://youtu.be/XfIlhJFmw3c?t=930)).